policy/release/slsa_build_scripted_build.rego

```diff
@@ -112,6 +113,37 @@ deny contains result if {
 	result := lib.result_helper(rego.metadata.chain(), [subject_image_ref])
 }
 
+# METADATA
+# title: Image built by trusted Task
+# description: >-
+#   Verify the digest of the image being validated is reported by a trusted Task in its IMAGE_DIGEST
+#   result.
+# custom:
+#   short_name: image_built_by_trusted_task
+#   failure_msg: 'Image %q not built by a trusted task: %s'
+#   solution: Make sure the build Pipeline definition uses a trusted Task to build images.
+#   collections:
+#   - redhat
+#   depends_on:
+#   - attestation_type.known_attestation_type
+#
+deny contains result if {
+	expected_digest := input.image.ref
+
+	# Find all the Tekton Bundle references from the Tasks that claim to have built the image being
+	# validated.
+	bundles := {bundle |
+		some attestation in lib.pipelinerun_attestations
+		some build_task in tkn.build_tasks(attestation)
+		digest := tkn.task_result(build_task, "IMAGE_DIGEST")
```

Comment:
Do we want to support [other types](https://tekton.dev/docs/chains/slsa-provenance/#output-artifacts) of results here as well?

Reply:
We should. But adding support here won't really help as the underlying utility functions, e.g. `tkn.build_tasks`, don't support them. Filed https://github.com/enterprise-contract/ec-policies/issues/840 so we can address this properly.